dlt/common/cli/runner/pipeline_script.py

```diff
@@ -0,0 +1,198 @@
+
+        Resources and sources must be initialized and bound beforehand.
+        """
+        members: PipelineMembers = {
+            "pipelines": {},
+            "sources": {},
+        }
+        for name, value in self.pipeline_module.__dict__.items():
+            # skip modules and private stuff
+            if isinstance(value, ModuleType) or name.startswith("_"):
+                continue
+
+            if isinstance(value, dlt.Pipeline):
+                members["pipelines"][name] = value
+
+            if isinstance(value, DltSource):
+                members["sources"][name] = value
+
+            if isinstance(value, DltResource):
+                if value._args_bound:
+                    members["sources"][name] = value
+                else:
+                    fmt.echo(fmt.info_style(f"Resource: {value.name} is not bound, skipping."))
+
+        return members
+
+    @property
+    def run_arguments(self) -> t.Dict[str, str]:
+        run_options: t.Dict[str, str] = {}
+        if not self.params.args:
+            return run_options
+
+        for arg in self.params.args:
+            arg_name, value = arg.split("=")
+            run_options[arg_name] = value
+
+        return run_options
+
+    def validate_arguments(self) -> None:
+        """Validates and checks"""
+        supported_args = {
+            "destination",
+            "staging",
+            "credentials",
+            "table_name",
+            "write_disposition",
+            "dataset_name",
+            "primary_key",
+            "schema_contract",
+            "loader_file_format",
+        }
+        errors = []
+        arguments = self.run_arguments
+        for arg_name, _ in arguments.items():
+            if arg_name not in supported_args:
+                errors.append(f"Invalid argument {arg_name}")
+
+        if (
+            "write_disposition" in arguments
+            and arguments["write_disposition"] not in self.supported_write_disposition
+        ):
+            errors.append(
+                "Invalid write disposition, select one of"
+                f" {'|'.join(self.supported_write_disposition)}"
+            )
+        if (
+            "loader_file_format" in arguments
+            and arguments["loader_file_format"] not in self.supported_formats
+        ):
+            errors.append(
+                f"Invalid loader file format, select one of {'|'.join(self.supported_formats)}"
+            )
+        if (
+            "schema_contract" in arguments
+            and arguments["schema_contract"] not in self.supported_evolution_modes
+        ):
+            errors.append(
+                "Invalid schema_contract mode, select one of"
+                f" {'|'.join(self.supported_evolution_modes)}"
+            )
+
+        if errors:
+            all_errors = "\n".join(errors)
+            raise RunnerError(f"One or more arguments are invalid:\n{all_errors}")
```

Comment:
@rudolfix I added this manual validation because imo it is not worth spending time to write more sophisticated auto inference and doscrings parsing to achieve it.